plugins/modules/ndo_interface_setting.py

```diff
@@ -0,0 +1,837 @@
+                match.details.pop("controls", None)
+
+            mso.sanitize(match.details)
+
+        else:
+            if not interface_type:
+                mso.fail_json(msg="Error: Missing required argument 'interface_type' for creating an Interface Policy Group.")
+            payload = {
+                "name": interface_policy_group,
+                "type": interface_type,
+                "templateId": mso_template.template.get("templateId"),
+                "schemaId": mso_template.template.get("schemaId"),
+                "llfc": {},
+                "stp": {},
+                "l2Interface": {},
+                "lldp": {},
+                "linkLevel": {},
+                "mcp": {},
+                "portChannelPolicy": {},
+            }
+
+            if description:
+                payload["description"] = description
+
+            if domains:  # domains is a list
+                domain_uuid = []
+                # get names and uuids of existing domains
+                existing_physical_domains = {domain["name"]: domain["uuid"] for domain in template_info.get("domains", [])}
+                existing_l3_domains = {domain["name"]: domain["uuid"] for domain in template_info.get("l3Domains", [])}
+                for item in domains:
+                    if item in existing_physical_domains:
+                        domain_uuid.append(existing_physical_domains[item])
+                    elif item in existing_l3_domains:
+                        domain_uuid.append(existing_l3_domains[item])
+                    else:
+                        mso.fail_json(msg="Domain '{0}' not found in the template '{1}'.".format(item, template))
+                payload["domains"] = domain_uuid
+
+            if sync_e:
+                # existing_sync_e contains names and uuids of existing SyncE policies
+                existing_sync_e = {sync_e["name"]: sync_e["uuid"] for sync_e in template_info.get("syncEthIntfPolicies", [])}
+                if sync_e in existing_sync_e:
+                    payload["syncEthPolicy"] = existing_sync_e[sync_e]
+                else:
+                    mso.fail_json(msg="SyncE policy '{0}' not found in the template '{1}'.".format(sync_e, template))
+
+            if access_macsec_policy:
+                # existing_access_macsec_policy contains names and uuids of existing SyncE policies
+                existing_access_macsec_policy = {
+                    access_macsec_policy["name"]: access_macsec_policy["uuid"] for access_macsec_policy in template_info.get("macsecPolicies", [])
+                }
+                if access_macsec_policy in existing_access_macsec_policy:
+                    payload["accessMACsecPolicy"] = existing_access_macsec_policy[access_macsec_policy]
+                else:
+                    mso.fail_json(msg="Access MACsec policy '{0}' not found in the template '{1}'.".format(access_macsec_policy, template))
+
+            # dictionaries
+            # cdp
```

Comment:
again think these comments do not really add value

Reply:
removed